specs/phase0/fork-choice.md

```diff
@@ -304,13 +366,69 @@ def store_target_checkpoint_state(store: Store, target: Checkpoint) -> None:
 
 ```python
 def update_latest_messages(store: Store, attesting_indices: Sequence[ValidatorIndex], attestation: Attestation) -> None:
-    target = attestation.data.target
+    attestation_slot = attestation.data.slot
     beacon_block_root = attestation.data.beacon_block_root
     for i in attesting_indices:
-        if i not in store.latest_messages or target.epoch > store.latest_messages[i].epoch:
-            store.latest_messages[i] = LatestMessage(epoch=target.epoch, root=beacon_block_root)
+        if i not in store.latest_messages or attestation_slot > store.latest_messages[i].slot:
```

Comment:
We check `is_valid_indexed_attestation()` (in `on_attestation()`) before this piece of code. This prevents the above case, since validators are assigned a single slot to attest for in each epoch.